tests/components/sensor/test_yr.py

```diff
@@ -32,12 +31,9 @@ def test_default_setup(self, betamax_session):
                    return_value=betamax_session):
             with patch('homeassistant.components.sensor.yr.dt_util.utcnow',
                        return_value=now):
-                assert sensor.setup(self.hass, {
-                    'sensor': {
-                        'platform': 'yr',
-                        'elevation': 0,
-                    }
-                })
+                    assert _setup_component(self.hass, 'sensor', {
```

Comment:
I found that if you do not add `self.hass.config.components = []` before the call to setup component, that the second call to `_setup_component` will just return success without validating the config because it believes it was already set up.
And because the ordering of tests is not predictable you have to put that in front of every call to `_setup_component`.